test/system/lifecycle/lifecycle_test.go

```diff
@@ -0,0 +1,281 @@
+			"keys":      base64.StdEncoding.EncodeToString(jwks),
+		}
+		err = f.RenderAndDeployTemplate(ctx, f.ShootClient, templates.CustomOIDCConfiguration, values)
+		Expect(err).ToNot(HaveOccurred())
+		getRequest := f.ShootClient.RESTClient().Get()
+		getRequest = getRequest.RequestURI("/apis/authentication.gardener.cloud/v1alpha1/openidconnects/custom")
+		getResponse := getRequest.Do(ctx)
+		Expect(getResponse.Error()).ToNot(HaveOccurred())
+
+		statusCode := 0
+		getResponse.StatusCode(&statusCode)
+		Expect(statusCode).To(Equal(200))
+
+		// Get token from seed
+		var ttl int64 = 1800
+		tokenReq, err := f.SeedClient.Kubernetes().CoreV1().ServiceAccounts("default").CreateToken(ctx, "default", &authenticationv1.TokenRequest{
+			Spec: authenticationv1.TokenRequestSpec{
+				Audiences:         []string{clientId},
+				ExpirationSeconds: &ttl,
+			},
+		}, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		expectedStatus := status{
+			Status:  "Failure",
+			Message: "forbidden: User \"custom-prefix:system:serviceaccount:default:default\" cannot get path \"/\"",
+			Reason:  "Forbidden",
+			Code:    403,
+		}
+
+		// Expect API calls to be authenticated
+		Eventually(func() status {
+			statusError, err := requestAPIServer(ctx, f.ShootClient.RESTConfig().CAData, f.ShootClient.RESTConfig().Host, tokenReq.Status.Token)
+			Expect(err).ToNot(HaveOccurred())
+
+			return status{
+				Code:    int(statusError.Code),
+				Message: statusError.Message,
+				Reason:  string(statusError.Reason),
+				Status:  statusError.Status,
+			}
+		}, time.Second*20, time.Second).Should(Equal(expectedStatus))
+
+		// Delete the oidc resource
+		deleteRequest := f.ShootClient.RESTClient().Delete()
+		deleteRequest = deleteRequest.RequestURI("/apis/authentication.gardener.cloud/v1alpha1/openidconnects/custom")
+		deleteResponse := deleteRequest.Do(ctx)
+		Expect(deleteResponse.Error()).ToNot(HaveOccurred())
+		getResponse.StatusCode(&statusCode)
+		Expect(statusCode).To(Equal(200))
+
+		// Expect API calls to be unauthorized
+		expectedStatus.Code = 401
+		expectedStatus.Message = "Unauthorized"
+		expectedStatus.Reason = string(metav1.StatusReasonUnauthorized)
+		Eventually(func() status {
+			statusError, err := requestAPIServer(ctx, f.ShootClient.RESTConfig().CAData, f.ShootClient.RESTConfig().Host, tokenReq.Status.Token)
+			Expect(err).ToNot(HaveOccurred())
+
+			return status{
+				Code:    int(statusError.Code),
+				Message: statusError.Message,
+				Reason:  string(statusError.Reason),
+				Status:  statusError.Status,
+			}
+		}, time.Second*20, time.Second).Should(Equal(expectedStatus))
+
+		// Disable the OIDC extension
+		err = f.UpdateShoot(ctx, func(shoot *gardencorev1beta1.Shoot) error {
+			newExtensions := make([]gardencorev1beta1.Extension, len(shoot.Spec.Extensions)-1)
+			for _, e := range shoot.Spec.Extensions {
+				if e.Type != constants.ExtensionType {
+					newExtensions = append(newExtensions, e)
+				}
```

Comment:
If the extension is globally enabled, you perhaps want to disable it with 
```golang
e.Disabled = pointer.Bool(true)
```

Reply:
I made it so that the initial extension configuration is preserved after the test execution.